pkg/controller/authentication/pgsql.go

```diff
@@ -0,0 +1,221 @@
+	}
+
+	return nil
+}
+
+func (r *ReconcileAuthentication) newPgsqlServiceSpec(instance *operatorv1alpha1.Authentication) *operatorv1alpha1.PgsqlServiceSpec {
+	storageClass := instance.Spec.PgsqlService.StorageClass
+
+	if storageClass == "" {
+		storageclass, err := r.getstorageclass()
+		if err != nil {
+			return nil
+		} else {
+			instance.Spec.PgsqlService.StorageClass = storageclass
+
+		}
+	} else {
+		scExist := r.storageclassAvailabe(storageClass)
+		if !scExist {
+			storageclass, err := r.getstorageclass()
+			if err != nil {
+				return nil
+			} else {
+				instance.Spec.PgsqlService.StorageClass = storageclass
+
+			}
+		} else {
+			// provided storageclass is present in the cluster.
+		}
+	}
+	if instance.Spec.PgsqlService.LogLevel == "" {
+		instance.Spec.PgsqlService.LogLevel = "info"
+	}
+	cpuReq := instance.Spec.PgsqlService.Resources.Requests.Cpu().String()
+	if cpuReq == "" {
+		cpuReq = "75m"
+	}
+	memReq := instance.Spec.PgsqlService.Resources.Requests.Memory().String()
+	if memReq == "" {
+		memReq = "256Mi"
+	}
+	esReq := instance.Spec.PgsqlService.Resources.Requests.StorageEphemeral().String()
+	if esReq == "" {
+		esReq = "128Mi"
+	}
+	cpuLim := instance.Spec.PgsqlService.Resources.Limits.Cpu().String()
+	if cpuLim == "" {
+		cpuLim = "200m"
+	}
+	memLim := instance.Spec.PgsqlService.Resources.Limits.Memory().String()
+	if memLim == "" {
+		memLim = "768Mi"
+	}
+	esLim := instance.Spec.PgsqlService.Resources.Limits.StorageEphemeral().String()
+	if esLim == "" {
+		esLim = "512Mi"
+
+	}
+	return &operatorv1alpha1.PgsqlServiceSpec{LogLevel: instance.Spec.PgsqlService.LogLevel, StorageClass: instance.Spec.PgsqlService.StorageClass, ImageName: shatag.GetImageRef("POSTGRESQL_IMAGE"), CpuReq: cpuReq, MemReq: memReq, ESReq: esReq, CpuLim: cpuLim, MemLim: memLim, ESLim: esLim}
+}
+
+func (r *ReconcileAuthentication) getstorageclass() (string, error) {
+	scList := &storagev1.StorageClassList{}
+	err := r.Reader.List(context.TODO(), scList)
+	if err != nil {
+		return "", err
+	}
+	if len(scList.Items) == 0 {
+		return "", fmt.Errorf("could not find storage class in the cluster")
+	}
+
+	var defaultSC []string
+	var nonDefaultSC []string
+
+	for _, sc := range scList.Items {
+		if sc.ObjectMeta.GetAnnotations()["storageclass.kubernetes.io/is-default-class"] == "true" || sc.ObjectMeta.GetAnnotations()["storageclass.beta.kubernetes.io/is-default-class"] == "true" {
+			defaultSC = append(defaultSC, sc.GetName())
+			continue
+		}
+		if sc.Provisioner == "kubernetes.io/no-provisioner" {
+			continue
+		}
+		nonDefaultSC = append(nonDefaultSC, sc.GetName())
+	}
+
+	if len(defaultSC) != 0 {
+		return defaultSC[0], nil
+	}
+
+	if len(nonDefaultSC) != 0 {
+		return nonDefaultSC[0], nil
+	}
+
+	return "", fmt.Errorf("could not find dynamic provisioner storage class in the cluster nor is there a default storage class")
+}
+
+func (r *ReconcileAuthentication) storageclassAvailabe(storageClass string) bool {
```

Comment:
```suggestion
func (r *ReconcileAuthentication) storageClassAvailable(storageClass string) bool {
```
Misspelled, and the casing is inconsistent (`storageClass` vs. `storageclass`.) In Go, tend to favor camelCase.

Reply:
Addressed